scripts/o3de/ExportScripts/export_standalone_monolithic_engine_centric.py

```diff
@@ -0,0 +1,160 @@
+mono_build_path     = (o3de_context.engine_path) / 'build' / 'mono' if args.mono_build_path is None else args.mono_build_path
+
+selected_platform = args.platform
+
+if not selected_platform:
+    o3de_logger.error("Unable to identify default host platform! Please supply the platform using '--platform'. Options are [pc, linux, mac].")
+    sys.exit(1)
+
+o3de_logger.info(f"Selected Project path: {o3de_context.project_path}")
+o3de_logger.info(f"Selected Engine path : {o3de_context.engine_path}")
+o3de_logger.info(f"Build path for non-monolithic executables: {args.non_mono_build_path}")
+o3de_logger.info(f"Build path for monolithic executables: {args.mono_build_path}")
+
+output_cache_path = args.output_path / 'Cache' / selected_platform
+os.makedirs(output_cache_path, exist_ok=True)
+
+# Build project assets and the engine (non-monolithic)
+if args.build_non_mono_tools:
+    process_command(['cmake', '-S', o3de_context.engine_path, '-B', str(non_mono_build_path), '-DLY_MONOLITHIC_GAME=0', f'-DLY_PROJECTS={o3de_context.project_path}'],
+                    cwd=o3de_context.engine_path)
+    process_command(['cmake', '--build', str(non_mono_build_path), '--target', 'AssetBundler', 'AssetBundlerBatch', 'AssetProcessor', 'AssetProcessorBatch', '--config','profile'],
+                    cwd=o3de_context.engine_path)
+    process_command(['cmake', '--build', str(non_mono_build_path), '--target', f'{project_name}.Assets', '--config', 'profile'],
+                    cwd=o3de_context.engine_path)
+
+# Build monolithic game
+process_command(['cmake', '-S', '.', '-B', str(mono_build_path), '-DLY_MONOLITHIC_GAME=1', '-DALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES=0', f'-DLY_PROJECTS={o3de_context.project_path}'],
+                cwd=o3de_context.engine_path)
+    
+if not args.no_game_launcher:
+    process_command(['cmake', '--build', str(mono_build_path), '--target', f'{project_name}.GameLauncher', '--config', args.config],
+                    cwd=o3de_context.engine_path) 
+
+if not args.no_server_launcher:
+    process_command(['cmake', '--build', str(mono_build_path), '--target', f'{project_name}.ServerLauncher', '--config', args.config],
+                    cwd=o3de_context.engine_path)
+
+if not args.no_unified_launcher:
+    process_command(['cmake', '--build', str(mono_build_path), '--target', f'{project_name}.UnifiedLauncher', '--config', args.config],
+                    cwd=o3de_context.engine_path)
+
+# Before bundling content, make sure that the necessary executables exist
+asset_bundler_batch_path = non_mono_build_path / 'bin' / 'profile' / ('AssetBundlerBatch' + ('.exe' if host_asset_platform=='pc' else ''))
+if not asset_bundler_batch_path.is_file():
+    o3de_logger.error(f"AssetBundlerBatch not found at path '{asset_bundler_batch_path}'. In order to bundle the data for project, this executable must be present!")
```

Comment:
Won't AssetBundlerBatch be available from the `cmake` build command above? If this is an expected file from the build, then an assert is more appropriate here. (In other words, we expect this executable to always exist, if it doesn't then there is something wrong in this workflow)

Reply:
There is the possibility that the user won't want to run build tools compilation, so that cmake command may not run. It is expected, so we can assert, but I chose this method since I want to provide more information to the user so they can pick the best recourse. That includes debug statements instructing users to run the build tool flag if it does turn out they are missing it. 

An assert's error trace clutters up the view in the logs imo, making it harder to read in this case.